Client/Frontend/Strings.swift

```diff
@@ -4,338 +4,586 @@
 
 import Foundation
 
+// MARK: - Localization bundle setup
+class BundleClass {}
+
 public struct Strings {
     public static let bundle = Bundle(for: BundleClass.self)
 }
 
-class BundleClass {}
+// MARK: - String last updated app version
+
+// Used as a helper enum to keep track of what app version strings were last updated in.
+fileprivate enum StringLastUpdatedAppVersion {
+    case v39
+
+    // Used for all cases before version 39.
+    case unknown
+}
 
-func MZLocalizedString(_ key: String, tableName: String? = nil, value: String = "", comment: String) -> String {
+// MARK: - Localization helper function
+fileprivate func MZLocalizedString(_ key: String, tableName: String? = nil, value: String = "", comment: String, lastUpdated: StringLastUpdatedAppVersion) -> String {
     return NSLocalizedString(key, tableName: tableName, bundle: Strings.bundle, value: value, comment: comment)
 }
 
+/// This file contains all strings for Firefox iOS.
+///
+/// As we continue to update strings, old strings may be present at the bottom of this
+/// file. To preserve a clean implementation of strings, this file should be organized
+/// alphabetically, according to specific screens or feature, on that screen. Each
+/// string should be under a struct giving a clear indication as to where it is being
+/// used. In this case we will prefer verbosity for the sake of accuracy, over brevity.
+/// Sub structs may, and should, also be used to separate functionality where it makes
+/// sense, but efforts should be made to keep structs two levels deep unless there are
+/// good reasons for doing otherwise.
+///
+/// Note that some strings belong to one feature that appears across mulitple screens
+/// throughout the application. An example is contextual hints. In this case, it makes
+/// more sense to organize all those strings under the specific feature.
+
+// MARK: - Alerts
+extension String {
+    public struct Alerts {
+
+    }
+}
+
+// MARK: - Browser View Controller
+extension String {
+    public struct BVC {
+        public struct General {
+
+        }
+
+        public struct MenuItems {
+            public struct Hamburger {
+
+            }
+
+            public struct ThreeDot {
+
+            }
+
+            public struct LongPressGesture {
+
+            }
+        }
+    }
```

Comment:
nit: cleanup and add them as needed

Reply:
So, for the empty structs - the reason was for screen organization as we talked about in that meeting. This is meant as a guide for how we decided to organize the app. So Iwanted to leave this in as an eforcement. If we remove now, who will remember and how will we know to enforce the organization of strings to make them cleaner.

Reply:
@electricRGB The fact that no one will remember is an issue. We should add a follow-up refactor PR to use these struct as leaving them empty hass no guarantee we'll remember using them. Hence my suggestion to clean up and add as needed. 

Its not best practice to leave empty struct assuming we *might* use them.

If you really would like to enforce them then I'd suggest adding atleast a few items and use these struct so we know what goes where and not leave empty. And make a ticket to further add more items in the struct as we go and remove empty ones if we don't need them.